src/masternode-sync.cpp

```diff
@@ -74,43 +83,36 @@ void CMasternodeSync::Process()
         */
         if(mnodeman.CountEnabled() == 0) {
             RequestedMasternodeAssets = MASTERNODE_SYNC_INITIAL;
-            RequestedMasternodeAttempt = 0;
-        }
-        return;
+        } else
+            return;
     }
 
-    if(c++ % MASTERNODE_SYNC_TIMEOUT != 0) return;
+    if(fDebug) LogPrintf("CMasternodeSync::Process() - tick %d RequestedMasternodeAssets %d\n", tick, RequestedMasternodeAssets);
 
-    if(fDebug) LogPrintf("CMasternodeSync::Process() - RequestedMasternodeAssets %d c %d\n", RequestedMasternodeAssets, c);
-
-    //request full mn list only if Masternodes.dat was updated quite a long time ago
     if(RequestedMasternodeAssets == MASTERNODE_SYNC_INITIAL) GetNextAsset();
 
-    CBlockIndex* pindexPrev = chainActive.Tip();
-    if(pindexPrev == NULL) return;
-
     LOCK(cs_vNodes);
     BOOST_FOREACH(CNode* pnode, vNodes)
     {
-        if (pnode->nVersion >= MIN_POOL_PEER_PROTO_VERSION)
-        {
-
-            if(RequestedMasternodeAssets == MASTERNODE_SYNC_SPORKS){
-                if(pnode->HasFulfilledRequest("getspork")) continue;
-                pnode->FulfilledRequest("getspork");
-
-                if(RequestedMasternodeAttempt <= 2){
-                    pnode->PushMessage("getsporks"); //get current network sporks
-                    if(RequestedMasternodeAttempt == 2) GetNextAsset();
-                    RequestedMasternodeAttempt++;
-                }
-                return;
+        if(RequestedMasternodeAssets == MASTERNODE_SYNC_SPORKS){
+            if(pnode->HasFulfilledRequest("getspork")) continue;
+            pnode->FulfilledRequest("getspork");
+
+            if(RequestedMasternodeAttempt <= 2){
+                pnode->PushMessage("getsporks"); //get current network sporks
+                if(RequestedMasternodeAttempt == 2) GetNextAsset();
+                RequestedMasternodeAttempt++;
             }
+            return;
+        }
 
-            //don't begin syncing until we're at a recent block
-            if(pindexPrev->nTime + 600 < GetTime()) return;
+        //don't begin syncing until we're almost at a recent block
+        if(pindexPrev->nHeight + 4 < pindexBestHeader->nHeight || pindexPrev->nTime + 600 < GetTime()) return;
```

Comment:
Oh, I just was reading it backwards somehow :) I think using && is probably
a lot better in that case

On Sun, Jul 19, 2015 at 7:46 AM, UdjinM6 <notifications@github.com> wrote:

> In src/masternode-sync.cpp
> <https://github.com/dashpay/dash/pull/441#discussion_r34958712>:
>
> >
> > -            //don't begin syncing until we're at a recent block
> > -            if(pindexPrev->nTime + 600 < GetTime()) return;
> > +        //don't begin syncing until we're almost at a recent block
> > +        if(pindexPrev->nHeight + 4 < pindexBestHeader->nHeight || pindexPrev->nTime + 600 < GetTime()) return;
>
> err... nope. Say we have 100 headers and only 10 blocks so
> if (10 + 4 < 100 || ...) return; // do not sync yet
> Or am I missing smth?
> I would agree however that it's better to use && here
> if(pindexPrev->nHeight + 4 < pindexBestHeader->nHeight &&
> pindexPrev->nTime + 600 < GetTime()) return;
> i.e. if block height is still low AND block time is low too THEN do
> nothing. Thoughts?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dashpay/dash/pull/441/files#r34958712>.
>